python/lib/scheduler/dmod/scheduler/job/job.py

```diff
@@ -1099,6 +1151,20 @@ def data_requirements(self, data_requirements: List[DataRequirement]):
         self._data_requirements = data_requirements
         self._reset_last_updated()
 
+    @property
+    def is_partitionable(self) -> bool:
+        """
+        Whether the requirements of the job support partitioning.
+
+        At this time, only NextGen model jobs can be partitioned.
```

Comment:
Would it make sense to try to tie certain job steps and state to a "model" extension and encapsulate them separately from the core scheduling attributes and job steps?  Maybe a job `pre` and `post` step that can a job extension can implement?

Reply:
It's an interesting thought.  

For this specifically, I believe the idea of a partitioning step does generalize to abstract job execution, even if the exact details for that step vary.  For a particular job (type) to not require or support partitioning is just a special case.  But dividing up the work for a job is so fundamental to one of the primary motives of DMOD (enough to get its own service created) that I think it belongs.

We will need to adjust the overall design for jobs and job execution, though.   I don't expect those to completely generalize as-is to evaluation and calibration workflows, and certainly not to the eventual workflows that will be desired beyond that.